tdms/src/argument_parser.cpp

```diff
@@ -0,0 +1,106 @@
+#include "stdexcept"
+#include "argument_parser.h"
+
+
+ArgumentParser::ArgumentParser() = default;
+
+ArgumentNamespace ArgumentParser::parse_args(int nargs, char *argv[]) {
+
+  auto args = ArgumentNamespace(nargs, argv);
+
+  if (args.have_flag("-h")){
+    print_help_message();
+    exit(0);
+  }
+
+  if( (args.num_non_flag != 2) &&  (args.num_non_flag != 3) ){
+    fprintf(stderr,"incorrect number of arguments\n");
```

Comment:
Any specific reason for not using `std::cerr`/`std::cout`?

Reply:
consistency with the rest of the codebase. I think it'd be nice to switch to configurable logging (as #12) in one fell swoop. happy to modify now if you have strong feelings though